derivation/derivation.go

```diff
@@ -0,0 +1,63 @@
+// nolint: govet
+package derivation
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type Derivation struct {
+	_       string  `@DerivationPrefix`
+	Content []Value `'('((@@ ','?)* @@?)? ')'`
+}
+
+type Value struct {
+	Outputs          []Output          `'[' ((@@ ','?)* @@? )']'','`
+	InputDerivations []InputDerivation `'[' ((@@ ','?)* @@? )?']'','`
+	InputSources     []string          `'[' ((@NixPath ','?)* @NixPath? )?']'','`
+	Platform         string            `@String ','`
+	Builder          string            `@String ','`
+	Arguments        []string          `'[' ((@String ','?)* (@NixPath|@String)? )?']'','`
+	EnvVars          []Env             `'[' ((@@ ','?)* (@@)? )']'`
+}
+
+type Output struct {
+	Content       string `'(' @String ','`
+	Path          string `@NixPath ','`
+	HashAlgorithm string `@String ','`
+	Hash          string `@String ')'`
+}
+
+type InputDerivation struct {
+	Path string   `'(' @NixPath ','`
+	Name []string `'[' ((@String ','?)* @String? )']' ')'`
+}
+
+type Env struct {
+	Key   string `'(' @String ','`
+	Value string `(@String|@NixPath)? ')'`
+}
+
+func ReadDerivation(filename string) (*Derivation, error) {
```

Comment:
I'd keep all file io outside of this function. Just accept an io.Reader as function argument, and call `parser.Parse(` with it, instead of `parser.ParseString` below.